src/main/java/org/zalando/nakadi/service/subscription/SubscriptionService.java

```diff
@@ -1,41 +1,199 @@
+import org.zalando.nakadi.domain.PaginationLinks;
 import org.zalando.nakadi.domain.Subscription;
+import org.zalando.nakadi.domain.SubscriptionBase;
 import org.zalando.nakadi.domain.SubscriptionEventTypeStats;
+import org.zalando.nakadi.domain.SubscriptionListWrapper;
 import org.zalando.nakadi.domain.TopicPartition;
+import org.zalando.nakadi.exceptions.DuplicatedSubscriptionException;
 import org.zalando.nakadi.exceptions.ExceptionWrapper;
+import org.zalando.nakadi.exceptions.InternalNakadiException;
 import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.exceptions.NoSuchSubscriptionException;
+import org.zalando.nakadi.exceptions.ServiceUnavailableException;
 import org.zalando.nakadi.repository.EventTypeRepository;
 import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.db.SubscriptionDbRepository;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.Result;
 import org.zalando.nakadi.service.subscription.model.Partition;
 import org.zalando.nakadi.service.subscription.zk.ZkSubscriptionClient;
 import org.zalando.nakadi.service.subscription.zk.ZkSubscriptionClientFactory;
+import org.zalando.nakadi.util.SubscriptionsUriHelper;
+import org.zalando.problem.MoreStatus;
+import org.zalando.problem.Problem;
 
+import javax.ws.rs.core.Response;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 @Component
 public class SubscriptionService {
 
+    private static final Logger LOG = LoggerFactory.getLogger(SubscriptionService.class);
+    private static final UriComponentsBuilder SUBSCRIPTION_PATH = UriComponentsBuilder.fromPath("/subscriptions/{id}");
+
+    private final SubscriptionDbRepository subscriptionRepository;
     private final EventTypeRepository eventTypeRepository;
     private final ZkSubscriptionClientFactory zkSubscriptionClientFactory;
     private final TopicRepository topicRepository;
 
     @Autowired
-    public SubscriptionService(final ZkSubscriptionClientFactory zkSubscriptionClientFactory,
+    public SubscriptionService(final SubscriptionDbRepository subscriptionRepository,
+                               final ZkSubscriptionClientFactory zkSubscriptionClientFactory,
                                final TopicRepository topicRepository,
                                final EventTypeRepository eventTypeRepository) {
+        this.subscriptionRepository = subscriptionRepository;
         this.zkSubscriptionClientFactory = zkSubscriptionClientFactory;
         this.topicRepository = topicRepository;
         this.eventTypeRepository = eventTypeRepository;
     }
 
-    public List<SubscriptionEventTypeStats> createSubscriptionStat(final Subscription subscription) {
+    public Result<Subscription> createSubscription(final SubscriptionBase subscriptionBase, final Client client)
```

Comment:
Usually a service layer is considered a domain layer. Returning HTTP entities doesn't make sense since it's specific to the outer controller layer. But creating all this separation seems artificial. That's why I don't care of having code in the controller.